src/parquet/column_reader.cc

```diff
@@ -212,6 +121,9 @@ bool ColumnReader::ReadNewPage() {
       // Read a data page.
       num_buffered_values_ = current_page_header_.data_page_header.num_values;
```

Comment:
The DataPageHeaderV2 counts the nulls separately. We could then use `num_buffered_values_ = num_values - num_nulls`